sdk/keyvault/keyvault-keys/src/lro/delete/operation.ts

```diff
@@ -0,0 +1,113 @@
+import { AbortSignalLike } from "@azure/abort-controller";
+import { PollOperationState, PollOperation } from "@azure/core-lro";
+import { RequestOptionsBase } from "@azure/core-http";
+import { DeletedKey, KeyClientInterface } from "../../keysModels";
+
+/**
+ * @interface
+ * An interface representing the state of a delete key's poll operation
+ */
+export interface DeleteKeyPollOperationState extends PollOperationState<DeletedKey> {
+  /**
+   * @member {string} [name] The name of the key that will be deleted
+   */
+  name: string;
+  /**
+   * @member {RequestOptionsBase} [requestOptions] The optional HTTP parameters that will be used to dele the key
+   */
+  requestOptions?: RequestOptionsBase;
+  /**
+   * @member {KeyClientInterface} [client] An instance of the key's client
+   */
+  client: KeyClientInterface;
+  /**
+   * @member {DeletedKey} [initialResponse] The initial response received the first time the service was reached by the operation's update function
+   */
+  initialResponse?: Error;
+  /**
+   * @member {DeletedKey} [previousResponse] The previous response received the last time the service was reached by the operation's update function
+   */
+  previousResponse?: Error;
+}
+
+/**
+ * @interface
+ * An interface representing a delete key's poll operation
+ */
+export interface DeleteKeyPollOperation
+  extends PollOperation<DeleteKeyPollOperationState, DeletedKey> {}
+
+/**
+ * @summary Reaches to the service and updates the delete key's poll operation.
+ * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+ */
+async function update(
+  this: DeleteKeyPollOperation,
+  options: {
+    abortSignal?: AbortSignalLike;
+    fireProgress?: (state: DeleteKeyPollOperationState) => void;
+  } = {}
+): Promise<DeleteKeyPollOperation> {
+  const state = this.state;
+  const { name, client } = state;
+
+  const requestOptions = state.requestOptions || {};
+  if (options.abortSignal) {
+    requestOptions.abortSignal = options.abortSignal;
+  }
+
+  if (!state.initialResponse) {
+    await client.deleteKey(name, requestOptions);
+    state.started = true;
```

Comment:
This needs to check if the DeletedKey which comes back has recoveryId set. if that's null or undefined possibly then the state is completed (no need to poll since soft delete is not enabled).

Reply:
Thank you!

Reply:
Is the current approach better?